drivers/modem/ublox-sara-r4.c

```diff
@@ -1258,6 +1348,41 @@ static int create_socket(struct modem_socket *sock, const struct sockaddr *addr)
 		return -1;
 	}
 
+	if (sock->ip_proto == IPPROTO_TLS_1_2) {
+		char buf[sizeof("AT+USECPRF=#,#,#######\r")];
+
+		snprintk(buf, sizeof(buf), "AT+USOSEC=%d,1,%d", sock->id, sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
+			       MDM_CMD_TIMEOUT);
+		/* Reset the security profile */
+		snprintk(buf, sizeof(buf), "AT+USECPRF=%d", sock->id);
+		modem_cmd_send(&mctx.iface, &mctx.cmd_handler, NULL, 0U, buf, &mdata.sem_response,
+			       MDM_CMD_TIMEOUT);
+		/* Disable server cert validation TODO:Validation level should be at least 1*/
```

Comment:
This looks like a serious issue to me, even for initial implementation. According to the documentation I've found, you don't need to register any root CA at all to successfully complete the handshake as the server certificate is ignored in this case. If we want to get this through, IMO it should be listed somewhere as a current limitation **in bold**.

Reply:
I've updated the code to set validation level to 1, which means according to manual:
> Level 1: certificate validation against a specific or a list of imported trusted root certificates.
Ideally, level 2 / 3 should be supported as well, but I can't afford to put more work into this right now.